### PR TITLE
Update @reach/router part for docs and examples

### PR DESCRIPTION
**Pull request by ng-hai**

### CODE REVIEW

File: examples/client-only-paths/package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "version": "1.0.0",
   "author": "Kyle Mathews <mathews.kyle@gmail.com>",
   "dependencies": {
+    "@reach/router": "^1.1.1",
```
Comment (KyleAMathews) [inline examples/client-only-paths/package.json]: We don't put @reach/router into peerDependencies so this isn't necessary

File: docs/docs/migrating-from-v1-to-v2.md [hunk 2]
```diff
@@ -412,8 +412,8 @@ const NewsFeed = () => (
 )
 
 const Photo = ({ location, photoId }) => {
-  if (location.state.fromFeed) {
-    return <FromFeedPhoto id={photoId} />
+  if (location.state.fromNewsFeed) {
```
Comment (KyleAMathews) [inline docs/docs/migrating-from-v1-to-v2.md]: Why was this changed?

Comment (ng-hai) [reply docs/docs/migrating-from-v1-to-v2.md]: Because the example shows that <Link /> in NewsFeed has `fromNewsFeed` state but the props that Photo received is `fromFeed`

```jsx
const NewsFeed = () => (
  <div>
    <Link to="photos/123" state={{ fromNewsFeed: true }} />
  </div>
)

const Photo = ({ location, photoId }) => {
  if (location.state.fromFeed) {
    return <FromFeedPhoto id={photoId} />
  } else {
    return <Photo id={photoId} />
  }
}
```
